spring-integration-jdbc/src/test/java/org/springframework/integration/jdbc/channel/PostgresChannelMessageTableSubscriberTests.java

```diff
@@ -268,7 +269,18 @@ public void testRenewConnection() throws Exception {
 		CountDownLatch latch = new CountDownLatch(2);
 		List<Object> payloads = new ArrayList<>();
 		CountDownLatch connectionLatch = new CountDownLatch(2);
-		connectionSupplier.onGetConnection = connectionLatch::countDown;
+		connectionSupplier.onGetConnection = conn -> {
+			connectionLatch.countDown();
+			new Thread(() -> {
+				try {
+					Thread.sleep(3000);
```

Comment:
I'm not very happy when we create unmanaged threads, even in tests.
Isn't there other way to achieve whatever is expected here?
I'm not sure, though, what is that we try to simulate with this separate thread...

Would you mind to elaborate the logic?

Thanks

Reply:
I added this to close the connection after 3 seconds so that the isValid() method returns false and the connection is renewed. Do you now a better way to wait and close the connection?

Reply:
Right. `3 seconds` is too much for our tests to perform.
We have like 5000 tests in the project. Now imagine if we don't care about the time on them.
How about just to have a connection to be closed immediately on a first call, so the next one would have it re-opened? 
Instead of new thread and 3 seconds delay just have an `AtomicBoolean` flag.

Reply:
I don't think using an atomic flag would work here ... As I want the same connection to first succeed on the `LISTEN` call and then fail after some time. I most probably don't get your suggestion, cause I don't see how to achieve this with an AtomicBoolean....

Reply:
Just changed your code to this and it still works:
```
		AtomicBoolean connectionCloseState = new AtomicBoolean();
		connectionSupplier.onGetConnection = conn -> {
			connectionLatch.countDown();
			if (connectionCloseState.compareAndSet(false, true)) {
				try {
					conn.close();
				}
				catch (Exception e) {
					//nop
				}
			}
		};
```
What do I miss?